packages/wrangler/src/__tests__/traverse-module-graph.test.ts

```diff
@@ -0,0 +1,181 @@
+import { mkdir, writeFile } from "fs/promises";
+import path from "path";
+import dedent from "ts-dedent";
+import traverseModuleGraph from "../traverse-module-graph";
+import { runInTempDir } from "./helpers/run-in-tmp";
+import type { ConfigModuleRuleType } from "../config";
+
+/*
+ * This file contains inline comments with the word "javascript"
+ * This signals to a compatible editor extension that the template string
+ * contents should be syntax-highlighted as JavaScript. One such extension
+ * is zjcompt.es6-string-javascript, but there are others.
+ */
+
+describe("traverse module graph", () => {
+	runInTempDir();
+	it("should not detect JS without module rules", async () => {
+		await writeFile(
+			"./index.js",
+			dedent/* javascript */ `
+			import { HELLO } from "./other.js"
+			export default {
+				async fetch(request) {
+					return new Response(HELLO)
+				}
+			}
+			`
+		);
+		await writeFile(
+			"./other.js",
+			dedent/* javascript */ `
+			export const HELLO = "WORLD"
+			`
+		);
+
+		const bundle = await traverseModuleGraph(
+			{
+				file: path.join(process.cwd(), "./index.js"),
+				directory: process.cwd(),
+				format: "modules",
+				moduleRoot: process.cwd(),
+			},
+			[]
+		);
+
+		expect(bundle.modules).toStrictEqual([]);
+	});
+	it.each([
+		["ESModule", "esm"],
+		["CommonJS", "commonjs"],
+	])("should detect JS as %s", async (type, format) => {
+		await writeFile(
+			"./index.js",
+			dedent/* javascript */ `
+			import { HELLO } from "./other.js"
+			export default {
+				async fetch(request) {
+					return new Response(HELLO)
+				}
+			}
+			`
+		);
+		await writeFile(
+			"./other.js",
+			dedent/* javascript */ `
+			export const HELLO = "WORLD"
+			`
+		);
+
+		const bundle = await traverseModuleGraph(
+			{
+				file: path.join(process.cwd(), "./index.js"),
+				directory: process.cwd(),
+				format: "modules",
+				moduleRoot: process.cwd(),
+			},
+			[{ type: type as ConfigModuleRuleType, globs: ["**/*.js"] }]
+		);
+
+		expect(bundle.modules[0].type).toStrictEqual(format);
+	});
+	it("should not resolve JS without module root", async () => {
```

Comment:
Out of curiosity... all the other tests in Wrangler have an empty line between `it()` blocks. But I notice you do not add them. Not blocking, but I think it would be good to be consistent. Is there a reason not to space these out?

Reply:
I wonder if a more accurate test description would be 


```suggestion
	it("should not resolve JS outside the module root", async () => {
```